### PR TITLE
20 backend routes

### PR DESCRIPTION
**Pull request by fadeomar**
Relates: #20 
prepare the server routes that contain 

 1. authentication routes 
 2. users routes 
 3. languages routes
 4. message routes 
 5. users routes
 6. errors routes

and require this files to the router

### CODE REVIEW

**Comment by Fatmasiam** [comment]
solve conflict @fadeomar

File: client/src/App.css [hunk 1]
```diff
@@ -0,0 +1,5 @@
+* {
+    padding: 0;
+    margin: 0;
+    box-sizing: border-box;
+}
```
Comment (Amoodaa) [inline client/src/App.css]: eol line here

Comment (Fatmasiam) [inline client/src/App.css]: why this file here ?!!!

File: client/src/components/pages/Login/index.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
+.loginPage {
```
Comment (Fatmasiam) [inline client/src/components/pages/Login/index.css]: and this file ?!!!!!!!!!!

**Comment by Fatmasiam** [comment]
please remove the unrelated file
@fadeomar  @yosefalnajjarofficial

File: server/controllers/auth/login.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+exports.login = (req, res) => {
```
Comment (Fatmasiam) [inline server/controllers/auth/login.js]: i think we deal to add the `next` for all routes
 so we can handle the error at once in the `error` middleware

File: server/controllers/auth/signup.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+exports.signup = (req, res) => {
+  res.send(' Hello from sginup file');
```
Comment (Fatmasiam) [inline server/controllers/auth/signup.js]: and here add next

File: server/controllers/users/getUsersByLang.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+exports.getUserByLang = (req, res, next) => {
```
Comment (Fatmasiam) [inline server/controllers/users/getUsersByLang.js]:
```suggestion
exports.getUsersByLang = (req, res, next) => {
```

File: server/controllers/users/getUserProfile.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+exports.getUserProfile = (req, res, next) => {
```
Comment (Fatmasiam) [inline server/controllers/users/getUserProfile.js]:
```suggestion
exports.getUserInfo = (req, res, next) => {
```

File: server/controllers/languages/language.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+// eslint-disable-next-line no-unused-vars
+exports.language = (req, res, next) => {
```
Comment (Fatmasiam) [inline server/controllers/languages/language.js]:
```suggestion
exports.getAllLanguages = (req, res, next) => {
```

#### I think we dealing to be constant on naming

File: server/controllers/users/getUserByInterest.js [hunk 8]
```diff
@@ -0,0 +1,4 @@
+// eslint-disable-next-line no-unused-vars
```
Comment (Fatmasiam) [inline server/controllers/users/getUserByInterest.js]:
```suggestion
```

please don't disable the ESlint on any file

Comment (Amoodaa) [reply server/controllers/users/getUserByInterest.js]: ahmad told us this one is fine 
they did it every where bc we didnt use next yet, when we implementation the function we will remove them dont worry :)

File: server/controllers/languages/language.js (hunk 7 above)
Comment (Amoodaa) [reply server/controllers/languages/language.js]: consistent* is the word fatma *^*

File: server/controllers/users/getUserByInterest.js (hunk 8 above)
Comment (Fatmasiam) [reply server/controllers/users/getUserByInterest.js]: anyways don't disable the eslint i know that we will use the next so we don't have an issue if the eslint told us that there is an error on it.
so the eslint error about `next` word does not make sense to me in this PR 
Are you get what i want to say !? @Amoodaa

File: server/controllers/languages/language.js (hunk 7 above)
Comment (Fatmasiam) [reply server/controllers/languages/language.js]: yeah @Amoodaa

File: client/src/App.css (hunk 1 above)
Comment (fadeomar) [reply client/src/App.css]: done

File: client/src/components/pages/Login/index.css (hunk 2 above)
Comment (fadeomar) [reply client/src/components/pages/Login/index.css]: done

File: server/controllers/users/getUsersByLang.js (hunk 5 above)
Comment (fadeomar) [reply server/controllers/users/getUsersByLang.js]: done

File: server/controllers/auth/signup.js (hunk 4 above)
Comment (fadeomar) [reply server/controllers/auth/signup.js]: done

File: server/controllers/auth/login.js (hunk 3 above)
Comment (fadeomar) [reply server/controllers/auth/login.js]: done

File: server/controllers/users/getUserByInterest.js (hunk 8 above)
Comment (fadeomar) [reply server/controllers/users/getUserByInterest.js]: done

File: server/controllers/languages/language.js (hunk 7 above)
Comment (fadeomar) [reply server/controllers/languages/language.js]: done

File: server/controllers/users/getUserProfile.js (hunk 6 above)
Comment (fadeomar) [reply server/controllers/users/getUserProfile.js]: done

**Comment by Fatmasiam** [comment]
@fadeomar  @yosefalnajjarofficial  
    Really its nice and helpful work. It's like excellent facilities and manages our work for the coming days 
  Thx

**Pull request merged**